File: java/link-message-client/src/main/java/link/message/client/content/complex/linear/GridLinearComplexMessageContentItem.java
package link.message.client.content.complex.linear;

import link.message.client.utils.Guard;


/**
 * Grid类型的线性消息内容项
 * 
 * @author zhongt
 *
 */
public class GridLinearComplexMessageContentItem extends LinearComplexMessageContentItem {
	public GridLinearComplexMessageContentItem() {
		this.displayType = LinearComplexMessageContentType.GRID;
	}
	
	public GridLinearComplexMessageContentItem(GridContent gridContent) {
		Guard.guardReqiredObject(gridContent, "gridContent must be set value.");
		if (gridContent.columns.size() != gridContent.columnCount) {
			Guard.throwIllegalArgumentException("gridContent's column count not equal columnCount.");
		}
		
		this.content     = gridContent;
		this.displayType = LinearComplexMessageContentType.GRID;
	}
}

File: net/Link.Message.Client/content/complex/linear/GridLinearComplexMessageContentItem.cs
using Link.Message.Client.utils;

namespace Link.Message.Client.content.complex.linear
{
	/// <summary>
	/// Grid类型的线性消息内容项
	/// </summary>
	public class GridLinearComplexMessageContentItem : LinearComplexMessageContentItem
	{
		public GridLinearComplexMessageContentItem()
		{
			DisplayType = LinearComplexMessageContentType.Grid;
		}

		public GridLinearComplexMessageContentItem(GridContent gridContent)
		{
			Guard.GuardReqiredObject(gridContent, "gridContent must be set value.");
			if (gridContent.columns.Count != gridContent.columnCount)
			{
				Guard.ThrowIllegalArgumentException("gridContent's column count not equal columnCount.");
			}

			Content = gridContent;
			DisplayType = LinearComplexMessageContentType.Grid;
		}
	}
}